lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -1432,11 +1432,16 @@ export abstract class CacheManager implements ICacheManager {
                     this.staticAuthorityOptions.canonicalAuthority
                 ) ||
                 this.staticAuthorityOptions.knownAuthorities;
-            if (
-                staticAliases &&
-                staticAliases.includes(environment) &&
-                staticAliases.includes(entity.environment)
-            ) {
+
+            const validEnvironment = staticAliases?.some((staticAlias) => {
```

Comment:
What was the bug? Is it related to the use "indexOf" instead of "includes" ?

Reply:
Why doesn't `includes` work here?
Is this related to managed identity? If not, I think it should be included in a separate PR into dev.

Reply:
These are the values during Managed Identity Flow:
```
staticAlias = [ 'https://login.microsoftonline.com/common/' ][ 'https://login.microsoftonline.com/common/' ]
environment:  login.microsoftonline.com
entity.environment:  https://login.microsoftonline.com/common/
```

The existing logic did not recognize that there is indeed a match here. The authority is fake, and so is the environment, which is why there is an issue here.